src/main/java/com/solace/spring/cloud/stream/binder/config/autoconfigure/SolaceJavaAutoConfiguration.java

```diff
@@ -0,0 +1,114 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.solace.spring.cloud.stream.binder.config.autoconfigure;
+
+import com.solace.spring.cloud.stream.binder.config.SolaceJavaProperties;
+import com.solace.spring.cloud.stream.binder.config.SolaceOAuthClientConfiguration;
+import com.solace.spring.cloud.stream.binder.oauth.SolaceSessionOAuth2TokenProvider;
+import com.solace.spring.cloud.stream.binder.oauth.SpringJCSMPFactory;
+import com.solacesystems.jcsmp.JCSMPChannelProperties;
+import com.solacesystems.jcsmp.JCSMPProperties;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.jms.JmsAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.lang.Nullable;
+
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.Set;
+
+@Configuration
+@AutoConfigureBefore(JmsAutoConfiguration.class)
+@ConditionalOnClass({JCSMPProperties.class})
+@ConditionalOnMissingBean(SpringJCSMPFactory.class)
+@EnableConfigurationProperties(SolaceJavaProperties.class)
+@Import(SolaceOAuthClientConfiguration.class)
+public class SolaceJavaAutoConfiguration {
+
+    private SolaceJavaProperties properties;
+
+    @Autowired
+    public SolaceJavaAutoConfiguration(SolaceJavaProperties properties) {
+        this.properties = properties;
+    }
+
+    /**
+     * Returns a {@link SpringJCSMPFactory} based on {@link JCSMPProperties} bean.
+     *
+     * @return {@link SpringJCSMPFactory} based on {@link JCSMPProperties} bean.
+     */
+    @Bean
+    public SpringJCSMPFactory getSpringJCSMPFactory(JCSMPProperties jcsmpProperties,
+                                                    @Nullable SolaceSessionOAuth2TokenProvider solaceSessionOAuth2TokenProvider) {
+        return new SpringJCSMPFactory(jcsmpProperties, solaceSessionOAuth2TokenProvider);
+    }
+
+    /**
+     * Returns a {@link JCSMPProperties} based on {@link com.solace.spring.cloud.stream.binder.config.SolaceJavaProperties}.
+     *
+     * @return {@link JCSMPProperties} based on {@link com.solace.spring.cloud.stream.binder.config.SolaceJavaProperties}.
+     */
+    @Bean
```

Comment:
did you test it with spring starter that injects the x509 certs? Is this still working?